packages/formstr-app/src/containers/ResponsesNew/components/AIAnalysisChat/index.tsx

```diff
@@ -0,0 +1,234 @@
+  const [isConnecting, setIsConnecting] = useState(false);
+  const [streamingText, setStreamingText] = useState('');
+  const initialAnalysisPerformed = useRef(false);
+  const messageListRef = useRef<HTMLDivElement>(null);
+  const trueDataRef = useRef<string>('');
+  const initialConnectionDone = useRef(false);
+
+  useEffect(() => {
+    if (messageListRef.current) {
+      messageListRef.current.scrollTop = messageListRef.current.scrollHeight;
+    }
+  }, [chatHistory, isAnalyzing, streamingText]);
+  
+  const fetchModels = useCallback(async () => {
+    setFetchingModels(true);
+    const result = await ollamaService.fetchModels();
+    if (result.success && result.models && result.models.length > 0) {
+        setAvailableModels(result.models);
+        const currentConfig = ollamaService.getConfig();
+        const modelStillExists = result.models.some(m => m.name === currentConfig.modelName);
+        if (modelStillExists) {
+            setSelectedModel(currentConfig.modelName);
+        } else {
+            setSelectedModel(result.models[0].name);
+            ollamaService.setConfig({ modelName: result.models[0].name });
+        }
+    } else {
+        setAvailableModels([]);
+    }
+    setFetchingModels(false);
+  }, []);
+
+  const testConnection = useCallback(async (showAlerts: boolean = false) => {
+      setIsConnecting(true);
+      const result = await ollamaService.testConnection();
+      if (result.success) {
+          if (showAlerts) message.success('Successfully connected to Ollama!');
+          setConnectionStatus(true);
+          fetchModels();
+      } else {
+          setConnectionStatus(false);
+          if (showAlerts) message.error(`Connection failed: ${result.error || 'Unknown error'}`);
+      }
+      setIsConnecting(false);
+  }, [fetchModels]);
+
+  const handleSend = () => {
+    if (!activePrompt.trim() || isAnalyzing) return;
+    setChatHistory(prev => [...prev, { sender: 'user', text: activePrompt }]);
+    setActivePrompt('');
+  };
+
+  useEffect(() => {
+    if (isVisible) {
+      if (!initialConnectionDone.current) {
+        trueDataRef.current = createAnalysisReport(responsesData, formSpec);
+        setSelectedModel(ollamaService.getConfig().modelName);
+        testConnection(false);
+        initialConnectionDone.current = true;
+      }
+    } else {
+        initialAnalysisPerformed.current = false;
+        initialConnectionDone.current = false;
+        setChatHistory([]);
+    }
+  }, [isVisible, responsesData, formSpec, testConnection]);
+
+  useEffect(() => {
+    const lastMessage = chatHistory[chatHistory.length - 1];
+    const isInitialCall = chatHistory.length === 0 && isVisible && !initialAnalysisPerformed.current && selectedModel && connectionStatus;
+    const isUserCall = lastMessage?.sender === 'user';
+
+    if (!isInitialCall && !isUserCall) return;
+
+    const runStreamingAnalysis = async () => {
```

Comment:
would be better if you can break this function down and move it out of the component somehow. Its too complex to understand what its doing